From 09d35e9b242434e85e0624a9978020f69a595b5f Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Tue, 13 Nov 2018 13:35:45 +0200
Subject: [PATCH 5111/5242] MLK-20189 sound: asoc: fix dereference before null
 check

commit  c262dde8ac10fb2a97a4a95e91c5e2cd48732fc0 from
https://source.codeaurora.org/external/imx/linux-imx.git

There may be a null pointer dereference, or else the comparison against
null is unnecessary.

Fixes Coverity issue CID5327850

Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index 7cce9da..72bffea 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -1421,10 +1421,10 @@ static int fsl_micfil_startup(struct snd_pcm_substream *substream,
 			      struct snd_soc_dai *dai)
 {
 	struct fsl_micfil *micfil = snd_soc_dai_get_drvdata(dai);
-	struct device *dev = &micfil->pdev->dev;
 
 	if (!micfil) {
-		dev_err(dev, "micfil dai priv_data not set\n");
+		dev_err(dai->dev,
+			"micfil dai priv_data not set\n");
 		return -EINVAL;
 	}
 
-- 
1.7.9.5

